File: app/src/main/java/com/ntsoftware/vspc/myvspc/MainActivity.java
package com.ntsoftware.vspc.myvspc;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.browser.customtabs.CustomTabsIntent;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration mAppBarConfiguration;

    private final Context context = this;

    @BindView(R.id.scroll_layout)
    CoordinatorLayout scroll_layout;

    @BindView(R.id.bottom_nav)
    BottomNavigationView navigationView;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.Theme_MyVSPC_NoActionBar);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);

        setupNavigation();

    }

    private void setupNavigation() {

        mAppBarConfiguration = new AppBarConfiguration.Builder(
                R.id.nav_discover,R.id.nav_news, R.id.nav_schedule, R.id.nav_other)
                .build();

        navController = Navigation.findNavController(this, R.id.nav_host_fragment);

        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);

        NavigationUI.setupWithNavController(navigationView, navController);

    }

    public CoordinatorLayout getScrollLayout() {
        return scroll_layout;
    }
}

File: app/src/main/java/com/ntsoftware/vspc/myvspc/MainActivity.kt
package com.ntsoftware.vspc.myvspc

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.textview.MaterialTextView
import com.ntsoftware.vspc.myvspc.storage.ScheduleCache

class MainActivity: AppCompatActivity() {

    private lateinit var app_bar_config: AppBarConfiguration

    private lateinit var navigation_view: BottomNavigationView

    private lateinit var toolbar: Toolbar

    lateinit var nav_controller: NavController

    lateinit var scroll_layout: CoordinatorLayout

    lateinit var scroll_layout_params: CoordinatorLayout.LayoutParams

    lateinit var scroll_layout_behavior: CoordinatorLayout.Behavior<CoordinatorLayout>

    val status_text_view: MaterialTextView by lazy {
        findViewById(R.id.screen_status)
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        setTheme(R.style.Theme_MyVSPC_NoActionBar)

        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        init()

        setSupportActionBar(toolbar)

        setupNavigation()
    }

    private fun setupNavigation() {
        app_bar_config = AppBarConfiguration.Builder(
                R.id.nav_vk, R.id.nav_instagram, R.id.nav_youtube, R.id.nav_other)
                .build()

        nav_controller = Navigation.findNavController(this, R.id.nav_host_fragment)

        NavigationUI.setupActionBarWithNavController(this, nav_controller, app_bar_config)

        NavigationUI.setupWithNavController(navigation_view, nav_controller)

    }

    private fun init() {
        scroll_layout = findViewById(R.id.scroll_layout)
        navigation_view = findViewById(R.id.bottom_nav)
        toolbar = findViewById(R.id.toolbar)

        scroll_layout_params = scroll_layout.layoutParams as CoordinatorLayout.LayoutParams

        if(scroll_layout_params.behavior != null) {
            scroll_layout_behavior = scroll_layout_params.behavior as CoordinatorLayout.Behavior<CoordinatorLayout>
        }

    }

    public fun needBehavior(on: Boolean) {
        if(on) {
            scroll_layout_params.behavior = scroll_layout_behavior
        }
        else {
            scroll_layout_params.behavior = null
        }
    }

}